### ISSUE TITLE
[iOS] Flutter apps are not showing up in iPhone Settings from iOS 13+

### ISSUE THREAD

**Issue by tomasbaran** [flutter/flutter#56358]
## Steps to Reproduce

1. Run flutter default app (the blue counter one)
2. Go to iPhone's Settings 

**Expected results:** 
You should see your app in iPhone's settings like this:

![IMG_8DFBA38776E1-1](https://user-images.githubusercontent.com/25743901/81058445-593c1500-8ec6-11ea-9e1c-99ab05854848.jpeg)

**Actual results:** 

Our Flutter is just not showing up.

![Simulator Screen Shot - iPhone 11 Pro - 2020-05-05 at 11 45 00](https://user-images.githubusercontent.com/25743901/81058589-a28c6480-8ec6-11ea-8ed7-6e2aa03bcb70.png)

Note: In iOS 12 it acts as it should, the issue is only with the newer iOS 13+ versions.

<details>
  <summary>Logs</summary>

```
```

```
```

```
[✓] Flutter (Channel master, 1.18.0-9.0.pre.122, on Mac OS X 10.15.4 19E287, locale es-ES)
    • Flutter version 1.18.0-9.0.pre.122 at /Users/tomas/Developer/flutter
    • Framework revision 9a99403ad2 (8 hours ago), 2020-05-04 22:29:02 -0400
    • Engine revision 95ecd9a405
    • Dart version 2.9.0 (build 2.9.0-5.0.dev e7bebea059)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/tomas/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.44.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.1

[✓] Connected device (4 available)
    • iPhone        • 00008020-000415840E50003A            • ios            • iOS 13.3.1
    • iPhone 11 Pro • A85E0A78-987E-41EA-9676-E33AC01D746D • ios            • com.apple.CoreSimulator.SimRuntime.iOS-13-4 (simulator)
    • Web Server    • web-server                           • web-javascript • Flutter Tools
    • Chrome        • chrome                               • web-javascript • Google Chrome 81.0.4044.129

• No issues found!
```

</details>

**Comment by VladyslavBondarenko** [comment]
Tried with my iOS 13.4.5 on current stable, beta, dev, master. The debug app is not shown in the list after deleting and installing anew. Before it was there, but I don't know for how long

**Comment by tomasbaran** [comment]
For iOS 14+ it doesn't show up in the iPhone Settings at all.

**Comment by tomasbaran** [comment]
Weird, in my case, it never shows up. Is it showing up the first time also in the Simulator @VladyslavBondarenko ?

**Comment by jmagman** [comment]
I don't see any of my debugging Flutter apps in Settings in my 10.3 simulator or on my real iOS 13.4 device.  Maybe iOS requires a [settings bundle](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/UserDefaults/Preferences/Preferences.html) to show up there?

What settings are you expecting to see for the default counter app?

**Comment by tomasbaran** [comment]
Well, first off, I believe all apps that are installed on iOS should be in iPhone's settings app by default like every other app on the App Store even when it doesn't use any permissions. Maybe for Siri additional settings.

But, especially those that use any of the permissions, e.g. location permissions. When I use https://pub.dev/packages/location_permissions or https://pub.dev/packages/geolocator and I also add the necessary lines of code to plist.info, the flutter app still wouldn't show up in the iPhone settings app. So, functions like `LocationPermissions().openAppSettings();` do not work.

**Comment by VladyslavBondarenko** [comment]
@tomasbaran no, it's not showing up in the settings neither on real device, nor on simulator.
But it was on the real device (after I deleted the app and tried to install anew) and probably for long time, it should be I installed it with some of the earlier versions and never deleted it

**Comment by tomasbaran** [comment]
Isn't this issue related to Siri & Dictation? Just thinking out loud since Siri&Dictation wasn't present before iOS13 came out.

**Comment by tomasbaran** [comment]
It looks like it's a weird bug that also Apple may at least partially responsible for: https://apple.stackexchange.com/questions/274921/why-are-my-app-settings-not-appearing-in-the-settings-app however, in Flutter's case, it works as it should in iOS 12 and never works in iOS 13. Killing nor restarting the iPhone doesn't solve the issue, so it also may be a completely unrelated issue.

**Comment by VladyslavBondarenko** [comment]
I can see  in settings my app, which I have run after update to iOS 13.5 on iPhone 7. Those installed before and not were not present before - still not present

**Comment by tomasbaran** [comment]
I just tested it on my end on iOS 13.5 both the default counter app as well as another app that I needed to access the locations settings in the first place. Neither of them got installed in the Settings app.

**Comment by ttlg** [comment]
I have same issue with real device:
iPhone XS Max 13.1.
Flutter stable.

**Comment by LukasP94** [comment]
Same issue with real device (iPhone 8)

**Comment by Gadda27** [comment]
Any news on this?

**Comment by florentin89** [comment]
Any news about this bug ?
I'm using the Flutter package "location 3.0.2" and my flutter app is not displayed in the Settings list of my physical device (iPhone 7 Plus with iOS 14.0.1).
Any help will be appreciated !

**Comment by tomasbaran** [comment]
@florentin89 if you want this issue to be resolved sooner, give it a thumb up on the original post, that's how Flutter teams will know it's an important issue. 

Cheers,

**Comment by jmagman** [comment]
I just tried with a new vanilla Xcode project, and it also doesn't show up in settings.  Flutter doesn't do anything particular about this, it's just an Xcode project that does enough work to embed the Flutter rendering frameworks, your compiled dart code, and load the first Flutter view.

You can open `ios/Runner.xcworkspace` and add whatever extensions, etc Apple documentation requires to get it to show up.

If you find some behavior that is specific to a newly `flutter create`d app that doesn't also happen with a newly created Xcode project, please let us know and we'll reopen.

**Comment by shr1k4nt** [comment]
Did anyone find a solution this issue? Tried with location: 3.2.4 and geolocator: 6.1.14, iOS 14.3 on physical device and on a simulator. 

![zsh](https://user-images.githubusercontent.com/4485699/106368252-94ca1980-6348-11eb-9aaa-7e6e73590fe4.png)

**Comment by tomasbaran** [comment]
I did NOT find any solution to this.

**Comment by shr1k4nt** [comment]
This is how I got it working on the device:

1. Reset Location & Privacy. (Not sure this helps but this is one of the steps I took)
2. For fetching current location used the following code from `geolocator: 6.1.14`.
`Geolocator.getCurrentPosition().then((value) => processCurrentPosition(value));`
3. This triggered the location permission prompt. After giving appropriate permission, I can now see my app in the Settings list and also in the Privacy>Location Services list.
4. After this both `Geolocator.openAppSettings()` and `Geolocator.openLocationSettings()` take me to the Settings > my app.

I hope this helps!
